medicines/doc-index-updater/src/delete_manager/mod.rs

```diff
@@ -247,6 +278,193 @@ mod test {
         );
     }
 
+    #[test]
+    fn failure_to_delete_blob_leaves_job_status_as_accepted() {
+        let mut state_manager = given_a_state_manager();
+        let mut removeable_message = given_we_have_a_delete_message();
+        let error = given_a_delete_blob_error();
+
+        block_on(handle_processing_error_for_delete_message(
+            &mut removeable_message,
+            error,
+            &state_manager,
+        ))
+        .unwrap();
+
+        assert_eq!(
+            state_manager.get_most_recently_set_status(),
+            JobStatus::Accepted
+        );
+    }
+
+    #[test]
+    fn failure_to_delete_blob_does_not_remove_message_from_service_bus() {
+        let state_manager = given_a_state_manager();
+        let mut removeable_message = given_we_have_a_delete_message();
+        let error = given_a_delete_blob_error();
+
+        block_on(handle_processing_error_for_delete_message(
+            &mut removeable_message,
+            error,
+            &state_manager,
+        ))
+        .unwrap();
+
+        assert_eq!(
+            removeable_message.remove_was_called, false,
+            "Removed message, but shouldn't"
+        );
+    }
+
+    #[test]
+    fn failure_to_restore_index_removes_message_since_cannot_be_retried() {
+        let state_manager = given_a_state_manager();
+        let mut removeable_message = given_we_have_a_delete_message();
+        let error = given_failure_to_restore_index();
+
+        block_on(handle_processing_error_for_delete_message(
+            &mut removeable_message,
+            error,
+            &state_manager,
+        ))
+        .unwrap();
+
+        assert_eq!(
+            removeable_message.remove_was_called, true,
+            "Didn't remove message, but should"
+        );
+    }
+
+    #[test]
+    fn failure_to_restore_index_leaves_job_status_as_error() {
+        let mut state_manager = given_a_state_manager();
+        let mut removeable_message = given_we_have_a_delete_message();
+        let error = given_failure_to_restore_index();
+
+        block_on(handle_processing_error_for_delete_message(
+            &mut removeable_message,
+            error,
+            &state_manager,
+        ))
+        .unwrap();
+
+        assert_eq!(
+            state_manager.get_most_recently_set_status(),
+            JobStatus::Error {
+                message: String::from("Cannot restore index for blob Blob Id: Error message"),
+                code: String::from(""),
+            },
+        );
+    }
+
+    #[test]
+    fn index_with_doc_returns_success() {
+        let removeable_message = given_we_have_a_delete_message().message;
+        let search_client = given_a_search_client_that_returns_results();
+        let storage_client = given_a_storage_client();
+        given_the_necessary_env_vars_are_initialised();
```

Comment:
You can only run this test now with make test which makes it a bit less flexible as the run test button in vscode won't work for it.

Is there any way around this? Is it totally necessary to use this env variable in the test, or more accurately, could this be injected into the function under test instead of hard wired to the env variable ?

Reply:
`given_the_necessary_env_vars_are_initialised` initialises the env vars necessary for this test to run, so removes manual prerequisite set-up. We're currently not using a testing framework that allows for set-up and tear-down, which would be a nice-to-have to make these steps cleaner. We could manually add a test wrapper or look at available testing crates as future steps.